### ISSUE TITLE
Run in remote: "cannot find operation"

### ISSUE THREAD

**Issue by albertotb** [guildai/guildai#140]
Test project with `guild.yml`:

```
- model: fashion
  description: Simple classifier for Fashion-MNIST
  operations:
    train:
      description: Train classifier
      main: train
```

Contents of `train.py`:

```
import numpy as np

x = 0.1
noise = 0.1

loss = (np.sin(5 * x) * (1 - np.tanh(x ** 2)) + np.random.randn() * noise)

print("loss: %f" % loss)
``` 

Locally it runs fine, (it does not show default value of flags as 0.6.6):

```
(base) alberto@serenity:~/guild-start$ guild run train
You are about to run fashion:train
Continue? (Y/n) y
loss: 0.391769
```

But when I run in the remote:

```
(base) alberto@serenity:~/guild-start$ guild run train --remote my_remote
You are about to run fashion:train on my_remote
Continue? (Y/n) y
Building package
package src: /home/alberto/guild-start
package dist: /tmp/guild-remote-stage-_ma6t0jw
running clean
removing 'build/lib' (and everything under it)
removing 'build/bdist.linux-x86_64' (and everything under it)
'build/scripts-3.7' does not exist -- can't clean it
removing 'build'
running bdist_wheel
running build
running build_py
package init file '/home/alberto/guild-start/__init__.py' not found (or not a regular file)
creating build
creating build/lib
creating build/lib/fashion
copying /home/alberto/guild-start/echo.py -> build/lib/fashion
copying /home/alberto/guild-start/train.py -> build/lib/fashion
copying /home/alberto/guild-start/guild.yml -> build/lib/fashion
installing to build/bdist.linux-x86_64/wheel
running install
running install_lib
creating build/bdist.linux-x86_64
creating build/bdist.linux-x86_64/wheel
creating build/bdist.linux-x86_64/wheel/fashion
copying build/lib/fashion/echo.py -> build/bdist.linux-x86_64/wheel/fashion
copying build/lib/fashion/guild.yml -> build/bdist.linux-x86_64/wheel/fashion
copying build/lib/fashion/train.py -> build/bdist.linux-x86_64/wheel/fashion
running install_egg_info
running egg_info
writing fashion.egg-info/PKG-INFO
writing dependency_links to fashion.egg-info/dependency_links.txt
writing entry points to fashion.egg-info/entry_points.txt
writing namespace_packages to fashion.egg-info/namespace_packages.txt
writing top-level names to fashion.egg-info/top_level.txt
reading manifest file 'fashion.egg-info/SOURCES.txt'
writing manifest file 'fashion.egg-info/SOURCES.txt'
Copying fashion.egg-info to build/bdist.linux-x86_64/wheel/fashion-0.0.0-py3.7.egg-info
running install_scripts
creating build/bdist.linux-x86_64/wheel/fashion-0.0.0.dist-info/WHEEL
creating '/tmp/guild-remote-stage-_ma6t0jw/fashion-0.0.0-py2.py3-none-any.whl' and adding 'build/bdist.linux-x86_64/wheel' to it
adding 'fashion/echo.py'
adding 'fashion/guild.yml'
adding 'fashion/train.py'
adding 'fashion-0.0.0.dist-info/METADATA'
adding 'fashion-0.0.0.dist-info/PACKAGE'
adding 'fashion-0.0.0.dist-info/WHEEL'
adding 'fashion-0.0.0.dist-info/entry_points.txt'
adding 'fashion-0.0.0.dist-info/namespace_packages.txt'
adding 'fashion-0.0.0.dist-info/top_level.txt'
adding 'fashion-0.0.0.dist-info/RECORD'
removing build/bdist.linux-x86_64/wheel
Initializing remote run
Copying package
sending incremental file list
fashion-0.0.0-py2.py3-none-any.whl

sent 2,641 bytes  received 35 bytes  1,784.00 bytes/sec
total size is 2,507  speedup is 0.94
Installing package and its dependencies
Processing ./fashion-0.0.0-py2.py3-none-any.whl
Installing collected packages: fashion
Successfully installed fashion-0.0.0
Starting fashion:train on my_remote as ca1d1ad02a9c44daafc6816a97fe63f8
bash: realpath: orden no encontrada
guild: cannot find operation fashion:train
Try 'guild operations' for a list of available operations.
```

Running `guild run fashion:train --remote my_remote` does not help either (same error)

**Comment by gar1t** [comment]
The Guild support for remote there is relying on `realpath` which is not available on the remote system. I'm marking this as a bug as Guild should work-around those cases.

In the meantime though, I think the fastest path to resolving your issue will be to make the `realpath` program available on the remote system.

What OS is that remote running?

**Comment by albertotb** [comment]
Thanks, `sudo apt install realpath` solved the issue. The remote is Ubuntu 14.04:

```
(base) my_user@my_remote:~$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 14.04.6 LTS
Release:        14.04
Codename:       trusty
```

**Comment by gar1t** [comment]
Great! Thanks for the report. We'll make sure this requirement on `realpath` is fixed in the next release.

**Comment by gar1t** [comment]
Sorry for the long delay! A fix for this will land in 0.7.3.

**Comment by gar1t** [comment]
I spoke too soon. We need to make `realpath` a requirement for ssh based remotes. Python can be made to work but it's quite a bit more complicated and I think `realpath` is a reasonable requirement. If anyone disagrees let's reopen and consider a Python command.

For the record, the issue is in supporting variable and user dir substitutions. The Python command, ala `$(python -c "import os; print(os.path.realpath(...)")` get's uncomfortably complicated to accomplish with `$(realpath ...)` does.
